src/corona_hakab_model/generation/matrix_generator.py

```diff
@@ -52,36 +57,29 @@ def __init__(
         self.matrix_consts = matrix_consts
         self._unpack_population_data(population_data)
         self.size = len(self.agents)
-        self.depth = len(ConnectionTypes)
 
-        CoronaMatrix = corona_matrix.get_corona_matrix_class(matrix_consts.use_parasymbolic_matrix)
-        self.logger.info("Using CoronaMatrix of type {}".format(CoronaMatrix.__name__))
-        self.matrix = CoronaMatrix(self.size, self.depth)
+        # todo allow generic matrix type
+        #CoronaMatrix = corona_matrix.get_corona_matrix_class(matrix_consts.use_parasymbolic_matrix)
+        #self.logger.info("Using CoronaMatrix of type {}".format(CoronaMatrix.__name__))
+        #self.matrix = CoronaMatrix(self.size, self.depth)
+
+        self.matrix = EncounterLayerSet()
+        self.matrix_type = "EncounterLayerSet"
 
         # create all sub matrices
-        with self.matrix.lock_rebuild():
-            # todo switch the depth logic, to get a connection type instead of int depth
-            current_depth = 0
-
-            for con_type in ConnectionTypes:
-                if con_type in Connect_To_All_types:
-                    self._create_fully_connected_circles_matrix(
-                        con_type, self.social_circles_by_connection_type[con_type], current_depth
-                    )
-                elif con_type in Random_Clustered_types:
-                    self._create_random_clustered_circles_matrix(
-                        con_type, self.social_circles_by_connection_type[con_type], current_depth
-                    )
-                elif con_type in Geographic_Clustered_types:
-                    self._create_community_clustered_circles_matrix(
-                        con_type, self.social_circles_by_connection_type[con_type], current_depth
-                    )
-                current_depth += 1
+        for con_type in ConnectionTypes:
+            if con_type in Connect_To_All_types:
+                self._create_fully_connected_layer(con_type, self.social_circles_by_connection_type[con_type])
```

Comment:
Why not save this functions in a dict (or named tuple) for each connection and then use comprehension instead of loops?

Reply:
i am not sure i understand what you mean.